### PR TITLE
Add available-cars page

### PR DESCRIPTION
**Pull request by Asbjoern09**

### CODE REVIEW

**Comment by Asbjoern09** [comment]
This is how it looks, i made it so it links to dynamically created car details pages, based off of car ID

image

File: native-wheels-frontend/app/(tabs)/available-cars.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      try {
+        const token = await AsyncStorage.getItem("token");
+        const response = await axios.get<Car[]>(
+          "http://localhost:8080/api/cars",
+          {
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+          }
+        );
```
Comment (jlars22) [inline native-wheels-frontend/app/(tabs)/available-cars.tsx]: I suggest separating the API call and type definition of `Car` from the page, similar to how the auth calls are structured.

By importing the `axios/instance.ts`, both the auth header and base URL are automatically set, simplifying API calls. For reference, check the `getUserInfo()` function in `axios/auth/api.ts` and the related `axios/auth/types.ts`.

Comment (Asbjoern09) [reply native-wheels-frontend/app/(tabs)/available-cars.tsx]: I agree, much better. I separated the api and type into a cars folder under `axios/cars`. I also separated the Car banner into a separate component in `components/cars` that maybe @Olsedyr can reuse for showing booked cars on the profile page.

**Review by jlars22 — APPROVED** [review]
Looks great!

**Pull request merged**